build-system/server/variable-substitution-tester.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let url;
+let variableSubstitution;
```

Comment:
So the idea is to have the server save those substitutions on one request, and serve them on other requests, like the "M+" button on a calculator? 🙃

Reply:
Yes, had trouble finding a way to get the parent doc to display the request info once <amp-analytics> (in the iframe) sent the request to the server. 

Do you know a better way to do this?

Reply:
I'm not sure, but this just doesn't feel like the right approach to me. Let's loop in @rsimha, who is more familiar with the dev server

Reply:
Sounds good, @rsimha let me know if you need me to explain what I'm attempting to do here

Reply:
There is precedent in our dev server to change its future behavior by visiting a URL that works as a switch. For example, this code:

https://github.com/ampproject/amphtml/blob/535a1e7caffcbbebd83d3a6d38ae0e43edcc8ad8/build-system/server/app.js#L97-L108

The difference here is that the URL params do not determine the logic or the content of the altered behavior. It's encoded in the server. Would that be possible here? Or are you trying to get the server to return an arbitrary response?

Reply:
Yes that is essentially what I am doing:

- `app.get('/run-variable-substitution', runVariableSubstitution)` runs the amp-analytics request (which gets stored in the server via the `variableSubstitution` and `url` variables)
- Then `app.get('/get-request', getRequest)` returns the `variableSubstitution` and `url` variables

Reply:
SGTM. No objections here.